### ISSUE TITLE
Could not fully ban all SHA256 usage in TLS

### ISSUE THREAD

**Issue by qunying** [openssl/openssl#9253]
For this test command:
```
#openssl ciphers -v 'HIGH:!SHA256:!DH:!DSS:!ECDSA:!CAMELLIA:!3DES:!SHA' | grep SHA256

TLS_CHACHA20_POLY1305_SHA256 TLSv1.3 Kx=any      Au=any  Enc=CHACHA20/POLY1305(256) Mac=AEAD
TLS_AES_128_GCM_SHA256  TLSv1.3 Kx=any      Au=any  Enc=AESGCM(128) Mac=AEAD
ECDHE-RSA-AES128-GCM-SHA256 TLSv1.2 Kx=ECDH     Au=RSA  Enc=AESGCM(128) Mac=AEAD
ECDHE-ARIA128-GCM-SHA256 TLSv1.2 Kx=ECDH     Au=RSA  Enc=ARIAGCM(128) Mac=AEAD
RSA-PSK-AES128-GCM-SHA256 TLSv1.2 Kx=RSAPSK   Au=RSA  Enc=AESGCM(128) Mac=AEAD
DHE-PSK-AES128-GCM-SHA256 TLSv1.2 Kx=DHEPSK   Au=PSK  Enc=AESGCM(128) Mac=AEAD
RSA-PSK-ARIA128-GCM-SHA256 TLSv1.2 Kx=RSAPSK   Au=RSA  Enc=ARIAGCM(128) Mac=AEAD
DHE-PSK-ARIA128-GCM-SHA256 TLSv1.2 Kx=DHEPSK   Au=PSK  Enc=ARIAGCM(128) Mac=AEAD
AES128-GCM-SHA256       TLSv1.2 Kx=RSA      Au=RSA  Enc=AESGCM(128) Mac=AEAD
ARIA128-GCM-SHA256      TLSv1.2 Kx=RSA      Au=RSA  Enc=ARIAGCM(128) Mac=AEAD
PSK-AES128-GCM-SHA256   TLSv1.2 Kx=PSK      Au=PSK  Enc=AESGCM(128) Mac=AEAD
PSK-ARIA128-GCM-SHA256  TLSv1.2 Kx=PSK      Au=PSK  Enc=ARIAGCM(128) Mac=AEAD
```
We still could see some SHA256 cipher suits available, should all the cipher suits with SHA256 be banned when it is disabled in the cipher list?

**Comment by davidben** [comment]
SHA256 only matches ciphers that use HMAC-SHA256 as the MAC, which is the legacy CBC-mode ciphers. It doesn't match the ciphers that only use SHA256 as the PRF function.

**Comment by qunying** [comment]
I understand that, but if I want to fully banned the SHA256 even in PRF part, could it be done?

**Comment by mattcaswell** [comment]
This is arguably a bug in the implementation of the SHA256 cipher group. However changing it to include ciphersuites that use SHA256 in the PRF function could be considered a breaking change. Alternatively we could treat this as a documentation bug.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input
